taxcalc/parameters.py

```diff
@@ -94,9 +94,6 @@ def set_default_vals(self, known_years=999999):
         """
         if hasattr(self, '_vals'):
             for name, data in self._vals.items():
-                if not isinstance(name, six.string_types):
-                    msg = 'parameter name {} is not a string'
-                    raise ValueError(msg.format(name))
```

Comment:
Paths to this section of code:

`Policy.__init__` --> `ParametersBase.initialize` --> `Parameters.set_default_vals`
`Behavior.__init__` --> `ParametersBase.initialize` --> `Parameters.set_default_vals`

In both of these cases, the values are read directly from the defaults specified in `current_law_policy.json` and `behavior.json`. Thus, it is impossible to have an incorrect name unless the name specified in the defaults files is a number or something like that. A test could potentially be added for this possibility.

`Policy.implement_reform` --> 
```
        self._validate_parameter_names_types(reform)
        ....
        # optionally apply cpi_offset to inflation_rates and re-initialize
        if Policy._cpi_offset_in_reform(reform):
            known_years = self._apply_reform_cpi_offset(reform)
            self.set_default_vals(known_years=known_years)
```
--> `ParametersBase.set_default_vals`

However, `Policy._validate_parameter_names_types` was called directly before `set_default_vals`. Thus, we can be certain that the names are strings and valid otherwise.